pallets/claim/src/tests.rs

```diff
@@ -436,6 +463,47 @@ fn cannot_add_beneficiaries_while_airdrop_inactive() {
     })
 }
 
+#[test]
+fn cannot_add_too_many_beneficiaries() {
+    test_with_configs(
+        WithGenesisBeneficiaries::Yes,
+        GenesisClaimBalance::Sufficient,
+    )
+    .execute_with(|| {
+        assert_noop!(
+            Claim::add_beneficiaries(
+                Origin::Signed(MANAGER_USER).into(),
+                utils::get_beneficiaries_map::<Test>(MaxBeneficiaries::get()).0
+            ),
+            Error::<Test>::TooManyBeneficiaries
+        );
+    })
+}
+
+#[test]
+fn cannot_add_beneficiaries_over_the_max() {
+    test().execute_with(|| {
+        assert_ok!(Claim::begin_airdrop(
+            Origin::Signed(MANAGER_USER).into(),
+            EMPTY_BENEFICIARIES_MAP.clone()
+        ));
+
+        let _ = Balances::mint_into(&Claim::account_id(), 1000000000).unwrap(); // Just to be safe
+        assert_ok!(Claim::add_beneficiaries(
+            Origin::Signed(MANAGER_USER).into(),
+            utils::get_beneficiaries_map::<Test>(MaxOpBeneficiaries::get()).0
+        ));
+
+        assert_noop!(
+            Claim::add_beneficiaries(
+                Origin::Signed(MANAGER_USER).into(),
+                NEW_BENEFICIARIES_MAP.clone()
```

Comment:
I would suggest testing that `MaxNumBeneficiariesReached` is raised as soon as you have `MaxBeneficiaries + 1` beneficiaries; maybe we can add less beneficiaries at line 494, or maybe add only 2 here?